plugins/utils/which-key.nix

```diff
@@ -11,6 +11,68 @@ let
   types = lib.nixvim.nixvimTypes;
 
   opt = options.plugins.which-key;
+
+  specExamples = [
+    # Basic group
```

Comment:
Is it worth using a `literalExpression` or `literalMD` example so that these comments are shown in the docs? I guess that wouldn't work for `settingsExample` since we can't _nest_ a literal expression _within_ a larger example :thinking:

Reply:
Yeah, it's a bit confusing in the `settingsExample` with a literalExpression. Can we parse the text attribute from it to get the actual nix ?

Reply:
If `settingsExample` was a literal expression we could include the string part of the "nested" literal, however TBH I don't think having a few comments is worth the effort?

We may also be able to do something with a ["pretty printer"](https://github.com/NixOS/nixpkgs/blob/47f999b551b9e7f515a678e326bd945ac7b7fda8/lib/generators.nix#L471-L474), assuming the docs enable `allowPretty`.

Reply:
But frankly I don't think it's worth the effort here

Reply:
> If `settingsExample` was a literal expression we could include the string part of the "nested" literal, however TBH I don't think having a few comments is worth the effort?

Yeah, I just saw the generated attribute set from passing a `literalExpression` to the `settingsExample` and got curious if we could grab the `text` attribute and convert it into a nix object easily. 

> But frankly I don't think it's worth the effort here

I agree, just got curious about how it could be done. :P The prettyprint example is interesting.

Reply:
We could have a helper similar to `literalExpression` that returns a pretty-printer that just prints the text? Only issue I see is the pretty-printer doesn't receive the current indentation string...